src/Flunt.Br/Document/Phone.cs

```diff
@@ -1,10 +1,192 @@
 using System.Text.RegularExpressions;
 using Flunt.Br.Document.interfaces;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Flunt.Br.Document
 {
     internal class Phone : IValidate
     {
-        public bool Validate(string value) => new Regex(@"^\(\d{2}\)\d{4}-\d{4}$").Match(value).Success;
+
+        public Phone()
+        {
+            this.format = null;
+        }
+        public Phone(string numberFormat)
```

Comment:
@marcoslcosta realmente essa validação precisava de um formato que pudesse ser passado por parâmetro, porém eu já havia começado uma implementação onde você possível ser flexível o formato, inclusive podendo definir o formato, sem precisar estar em uma lista de formatos pré-definidos. A funcionalidade de definir um formato já está pronta e funcionando com cobertura de testes, mas acabei deixando parado para uma situação de precisar definir uma caractere opcional como no caso de celular, você pode dar uma olhada na [branch](https://github.com/lira92/flunt.br/compare/1-suportar-mais-telefones?expand=1) ou na issue https://github.com/lira92/flunt.br/issues/1

Reply:
Se você observar, toda a flexibilidade está no construtor da classe, eu criei duas sobrecargas no metodo 
IsPhone(this Contract contract, string value, string property, string message)
IsPhone(this Contract contract, string value, string numberFormat, string property, string message)

Dando a flexibilidade de passar o formato do telefone desejado, porém se não houver valor passado, ele irá procurar se o telefone é valido em uma lista de formatos

Reply:
sim eu compreendo, vou baixar o código pra dar uma olhada como seria a utilização, mas dessa forma, caso algum formato que a lib não dê suporte é necessário adicionar na lista de formatos suportados certo?

Reply:
@marcoslcosta vou baixar pra dar uma olhada, como seria a utilização.